File: src/main/java/com/example/CargoTracking/dto/InternationalRoadReportPerformance.java
package com.example.CargoTracking.dto;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class InternationalRoadReportPerformance {
    private Long id;
    private String preAlertNumber;
    private String referenceNumber;
    private String origin;
    private String destination;
    private String route;
    private String vehicleType;
    private LocalDateTime actualTimeDeparture;
    private LocalDateTime actualTimeArrival;
    private LocalDateTime offloaded;
    private Long totalTransitTime;
    private Long totalLeadTime;

}

File: frontend/src/app/model/InternationalRoadReportPerformance.ts
export interface InternationalRoadReportPerformance {
  id:number|null;
  preAlertNumber: string | null;
  referenceNumber: string | null;
  origin: string | null;
  destination: string | null;
  route: string | null;
  vehicleType: string | null;
  actualTimeDeparture: Date | string | null;
  actualTimeArrival: Date | string | null;
  offloaded: Date | string | null;
  totalTransitTime: number | null;
  totalLeadTime: number | null;
}
